app/assets/scripts/utils/map-utils.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 import { get } from 'object-path';
 import getCentroid from '@turf/centroid';
-import * as governorates from './governorates';
+import * as governoratesMeta from './governorates';
+import * as districtsMeta from './districts';
 import { isOntime } from '../components/project-card';
 
-module.exports.GOVERNORATE = 'egy2';
-module.exports.DISTRICT = 'districts';
+const GOVERNORATE = 'egy2';
+const DISTRICT = 'districts';
+module.exports.GOVERNORATE = GOVERNORATE;
+module.exports.DISTRICT = DISTRICT;
+
+// helps to find fallback governorate when district included but missing from spatial data
+// http://codereview.stackexchange.com/questions/73714/find-a-nested-property-in-an-object
+function findFallback (o, fallbackId) {
+  if (o.fallback === fallbackId) {
+    return o;
+  }
+  var result, p;
+  for (p in o) {
+    if (o.hasOwnProperty(p) && typeof o[p] === 'object') {
+      result = findFallback(o[p], fallbackId);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return result;
+}
```

Comment:
The structure is going to be using the district ID as keys, which is another reason the `badId` needs to be stored (in order to be able to remove this function). I changed `badId` to `districtId` to clarify.